### ISSUE TITLE
is BlockGrad symbol equal to setting propagate_down as false in caffe?

### ISSUE THREAD

**Issue by chenjx1005** [dmlc/mxnet#3073]
In caffe if I set propagate_down as false in some middle layer in my net, I can speed up the back_ward computing time. And this is useful when finetuning I think. 
I tried BlockGrad symbol in mxnet like this:
`block = mx.sym.BlockGrad(top_layer, name=top_layer.name+"_block")`
`myfc = mx.sym.FullyConnected(block, num_hidden=args.num_classes, name='myfc')`
but I did not see any speedup.
Is there some way to speed up the finetuning like setting propagate_down in caffe?

**Comment by piiswrong** [comment]
It only set the gradient to 0. If you don't want to compute the gradient, don't bind the corresponding gradient array.

**Comment by winstywang** [comment]
No, NEVER USE BLOCKGRAD IF YOU HAVE WEIGHT DECAY!!! It only makes the gradient from higher layer to 0, but the weight decay will drive the weight to 0 gradually.

**Comment by chenjx1005** [comment]
@piiswrong @winstywang  thank you for quick replies. I will try not binding the corresponding gradient array.

**Comment by chenjx1005** [comment]
@piiswrong  not binding the gradient arrays seems impossible with APIs in FeedForward model. Can you tell me the way to un-bind the gradient arrays of a FeedForward model? Thank you.
